BlocBrowser/BlocBrowser/AwesomeFloatingToolbar.m

```diff
@@ -0,0 +1,163 @@
+//
+
+#import "AwesomeFloatingToolbar.h"
+
+#define color1 [UIColor colorWithRed:199/255.0 green:158/255.0 blue:203/255.0 alpha:1]
+#define color2 [UIColor colorWithRed:255/255.0 green:105/255.0 blue:97/255.0 alpha:1]
+#define color3 [UIColor colorWithRed:222/255.0 green:165/255.0 blue:164/255.0 alpha:1]
+#define color4 [UIColor colorWithRed:255/255.0 green:179/255.0 blue:71/255.0 alpha:1]
+
+@class ViewController;
+
+@interface AwesomeFloatingToolbar ()
+
+@property (nonatomic, strong) NSArray *currentTitles;
+@property (nonatomic, strong) NSMutableArray *colors;
+@property (nonatomic, strong) NSArray *labels;
+@property (nonatomic, weak) UIButton *currentLabel;
+@property (nonatomic, strong) UIPanGestureRecognizer *panGesture;
+@property (nonatomic, strong) UIPinchGestureRecognizer *pinchGesture;
+@property (nonatomic, strong) UILongPressGestureRecognizer *longPressGesture;
+
+@end
+
+@implementation AwesomeFloatingToolbar
+
+- (instancetype) initWithFourTitles:(NSArray *)titles {
+    self = [super init];
+    
+    if (self) {
+        self.currentTitles = titles;
+        
+        self.colors = [@[color1, color2, color3, color4] mutableCopy];
+        
+        NSMutableArray *labelArray = [[NSMutableArray alloc] init];
+        for (NSString *currentTitle in self.currentTitles) {
+            UIButton *label = [[UIButton alloc] init];
+            
+            label.userInteractionEnabled = NO;
+            label.alpha = 0.25;
+            
+            NSUInteger currentTitleIndex = [self.currentTitles indexOfObject:currentTitle];
+            NSString *titleForThisLabel = [self.currentTitles objectAtIndex:currentTitleIndex];
+            
+            label.titleLabel.textAlignment = NSTextAlignmentCenter;
+            label.titleLabel.font = [UIFont systemFontOfSize:15];
+            label.titleLabel.textColor = [UIColor whiteColor];
+            [label setTitle:titleForThisLabel forState:UIControlStateNormal];
+            
+            [label addTarget:self action:@selector(buttonTapped:) forControlEvents:UIControlEventTouchUpInside];
+            
+            [labelArray addObject:label];
+        }
+        self.labels = labelArray;
+        [self rotateColorAndAssign];
+        
+        for (UIButton *label in self.labels) {
+            [self addSubview:label];
+        }
+        
+        // gesture recognizer setup
+        self.panGesture = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(panFired:)];
+        [self addGestureRecognizer:self.panGesture];
+        self.pinchGesture = [[UIPinchGestureRecognizer alloc] initWithTarget:self action:@selector(pinchFired:)];
+        [self addGestureRecognizer:self.pinchGesture];
+        self.longPressGesture = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(longPressFired:)];
+        [self addGestureRecognizer:self.longPressGesture];
+    }
+    
+    return self;
+}
+
+- (void) layoutSubviews {
```

Comment:
So even with all of the changes I've suggested already, our toolbar won't scale like we want it to... so what are we fighting against?

Well, each time our `UIView`s change UIKit has to redraw them... or lay them out... so `layoutSubviews` gets called... a *lot*.
We only want to do this initial button/label layout once... which is an excellent case for using `dispatch_once`...
I would add the following before our code, and enclose it with `});`, and you should be good to go:
```
    static dispatch_once_t onceToken;
    dispatch_once( &onceToken, ^{
```